### PR TITLE
[Modules] Azure Firewall change name of first IP config

### PR DESCRIPTION
**Pull request by rahalan**
# Description

change of name of IP config (with the old name the deployment could fail)

| Pipeline |
| - |
|[Network: AzureFirewalls](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.azurefirewalls.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.Network/azureFirewalls/deploy.bicep [hunk 1]
```diff
@@ -150,7 +150,7 @@ var newPip = {
 
 var ipConfigurations = concat([
     {
-      name: 'IpConfAzureFirewallSubnet'
+      name: 'pip-${name}-default'
```
Comment (MrMCake) [inline modules/Microsoft.Network/azureFirewalls/deploy.bicep:153]: Out of curiosity: Did it fail because of conflicts?

Comment (rahalan) [reply modules/Microsoft.Network/azureFirewalls/deploy.bicep:153]: Don't know. After 2h I got the error: "error occured". That change fixed it.

Comment (ahmadabdalla) [reply modules/Microsoft.Network/azureFirewalls/deploy.bicep:153]: I am not sure if that would solve the issue. But I would suggest the name here to not to be the FW name. But the the public IP name in there as well. e.g.: 
`IpConf-pip-adp-carml-az-pip-x-fw` 
and not have the word 'default'

Comment (MrMCake) [reply modules/Microsoft.Network/azureFirewalls/deploy.bicep:153]: Agreed

Comment (rahalan) [reply modules/Microsoft.Network/azureFirewalls/deploy.bicep]: @MrMCake I changed it as discussed, [Network: AzureFirewalls](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.azurefirewalls.yml)

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**